src/main/java/com/hubspot/smtp/client/SmtpSession.java

```diff
@@ -142,6 +149,89 @@ public boolean isEncrypted() {
     return channel.pipeline().get(SslHandler.class) != null;
   }
 
+  public CompletableFuture<SmtpClientResponse[]> send(String from, String to, MessageContent content) {
+    Preconditions.checkNotNull(from);
+    Preconditions.checkNotNull(to);
+    Preconditions.checkNotNull(content);
+    checkMessageSize(content.size());
+
+    if (ehloResponse.isSupported(Extension.CHUNKING)) {
+      return sendAsChunked(from, to, content);
+    }
+
+    if (content.getEncoding() == MessageContentEncoding.SEVEN_BIT) {
+      return sendAs7Bit(from, to, content);
+    }
+
+    if (ehloResponse.isSupported(Extension.EIGHT_BIT_MIME)) {
+      return sendAs8BitMime(from, to, content);
+    }
+
+    if (content.count8bitCharacters() == 0) {
```

Comment:
do you need this api? would a better API be `boolean has8BitCharacters()`? That would allow short circuit evaluation.